src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Device;
+using System.Device.I2c;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// Binding that exposes an INA219 Bidirectional Current/Power monitor.
+    /// 
+    /// The INA219 is a high-side current shunt and power monitor with an I2C interface.
+    /// The INA219 monitors both shunt drop and supply voltage, with programmable conversion
+    /// times and filtering. A programmable calibration value, combined with an internal multiplier,
+    /// enables direct readouts in amperes. An additional multiplying register calculates power in watts.
+    /// <see href="http://www.ti.com/lit/ds/symlink/ina219.pdf"/>
+    /// </summary>
+    public partial class Ina219 : IDisposable
+    {
+        private readonly Dictionary<PgaSensitivity, float> _sensitivityVoltage = new Dictionary<PgaSensitivity, float>() { {PgaSensitivity.PlusOrMinus40mv, 0.04F }, { PgaSensitivity.PlusOrMinus80mv, 0.08F }, { PgaSensitivity.PlusOrMinus160mv, 0.16F }, { PgaSensitivity.PlusOrMinus320mv, 0.32F } };
+        private readonly Dictionary<AdcResolutionOrSamples, int> _readDelays = new Dictionary<AdcResolutionOrSamples, int>() { { AdcResolutionOrSamples.Adc9Bit, 84 }, { AdcResolutionOrSamples.Adc10Bit, 148 }, { AdcResolutionOrSamples.Adc11Bit, 276 }, { AdcResolutionOrSamples.Adc12Bit, 532 },
+          {AdcResolutionOrSamples.Adc2Sample, 1006}, {AdcResolutionOrSamples.Adc4Sample, 2130}, {AdcResolutionOrSamples.Adc8Sample, 4260}, {AdcResolutionOrSamples.Adc16Sample, 8510}, {AdcResolutionOrSamples.Adc32Sample, 17020}, {AdcResolutionOrSamples.Adc64Sample, 34050}, {AdcResolutionOrSamples.Adc128Sample, 68100}};
+        private I2cDevice _i2cDevice;
+        private bool _disposeI2cDevice = false;
+        private ushort _calibrationValue;
+        private AdcResolutionOrSamples _busAdcResSamp;
+        private AdcResolutionOrSamples _shuntAdcResSamp;
+
+        /// <summary>
+        /// The CurrentLsb property specifies the electrical current in amperes respresented by the least significant bit of the INA219 Current register.
+        /// It is used by the GetCurrent method to calculate the current from the value exposed by the current register
+        /// <see cref="GetCurrent"/>
+        /// </summary>
+        /// <remarks>
+        /// This property defaults to 1. It is set by calling SetCalibration with values for the ADC sensitivity, current range and optionally shunt resistance.
+        /// If the value of the calibration register is set manually then this property will also need to be set for GetCurrent to return the correct value.
+        /// <see cref="SetCalibration(PgaSensitivity, float, float)"/> <seealso cref="SetCalibration(ushort)"/>
+        /// </remarks>
+        public float CurrentLsb { get; set; }
```

Comment:
Lsb should not be in the name since it's transparent to user. I'd go with name something like `CurrentVoltage`.

Also it probably should not be settable (private set is ok)

Reply:
Other thing is that doc should mention the unit, i.e. V / mV / relative to Vcc

Reply:
don’t really agree as it is the amount of current represented by the least significant bit of the current register. Perhaps it’s CurrentLsbVoltage. It is not private as the users has the option of using the default calibration by calling set calibration with the full scale current value or setting the values directly to optimal values for the applications and components installed. In essence the current calibration is a combination Of the calibration register and the CurrentLsb property. If you wanted less fine grained control then you could just use the calibration register but the data sheet recommends setting the calibration register to the number which gives the best add resolution and then scaling using CurrentLsb. Hope this makes some sense

Reply:
If this is just partial sensor reading I'd prefer if we made this private and instead exposed some API which allows you to read the actual current

Reply:
It’s a partial calibration value used alongside the calibration register and so it may need to be user set. I can see how I could make the functionality part of the set calibration method. Sometime I just need to step back from the issue.